### PR TITLE
launchd - array instead of a hash for Apple's associated_bundle_ident…

### PR DESCRIPTION
**Pull request by Arequ**
Should address the issue where the launchd.rb resource requires a hash instead of an array per Apple's docs. 
 #13476

### CODE REVIEW

File: lib/chef/resource/launchd.rb
Comment (tpowell-progress) [inline lib/chef/resource/launchd.rb:132]: Seems like this needs to at a minimum be isolated to only macOS platforms. Would also like to see some testing around this to ensure that it's being handled correctly for macOS and non-macOS platforms

**Comment by tpowell-progress** [comment]
Approved the workflows to see if there's any current breakage, but need to revisit.

**Review by tpowell-progress — CHANGES_REQUESTED** [review]
Add tests and verify that non-macOS platforms aren't broken.

**Comment by Arequ** [comment]
> Add tests and verify that non-macOS platforms aren't broken.

This resource should only be applicable to macos per the documentation so I went ahead and restricted it to the darwin platform. I also added a kitchen test to verify that it causes an exception on Linux as it's supposed to. 

The kitchen actions are failing due to other resources and they can be observed failing in the most recent main branch commit. Please review the failed kitchen run and confirm this change is related.

I tried removing the line that was creating the error in the linux end-to-end test but it produced another error in the resulting action. Somethings going on with the way the Amazon Linux container is being spun up and tested on.

I'm not sure how to address the existing issues but this change should not introduced any additional problems. In its current state, it is creating plist files on macOS Ventura that are not valid.

Please advise on how to proceed, I'd like to continue using this resource on macOS.

File: README.md
Comment (jaymzh) [inline README.md:16]: remove this?

**Comment by tpowell-progress** [comment]
@Arequ Please rebase on `main` to resolve the conflict.

File: README.md
Comment (Arequ) [reply README.md]: done

**Comment by Arequ** [comment]
> @Arequ Please rebase on `main` to resolve the conflict.

@tpowell-progress, completed. Please pardon that I missed the DCO/obvious fix tagging. I would probably classify this as an "obvious fix". Should I make any changes to include this?

**Comment by tpowell-progress** [comment]
> > @Arequ Please rebase on `main` to resolve the conflict.
> 
> @tpowell-progress, completed. Please pardon that I missed the DCO/obvious fix tagging. I would probably classify this as an "obvious fix". Should I make any changes to include this?

@Arequ Looks like you lost the DCO on the rebase.

**Comment by tpowell-progress** [comment]
> > > @Arequ Please rebase on `main` to resolve the conflict.
> > 
> > 
> > @tpowell-progress, completed. Please pardon that I missed the DCO/obvious fix tagging. I would probably classify this as an "obvious fix". Should I make any changes to include this?
> 
> @Arequ Looks like you lost the DCO on the rebase.

@Arequ, I misread your comment, but I think we want DCO on this.

**Comment by Arequ** [comment]
updated

**Comment by erikng** [comment]
Can we get this merged? It should have been a one line change and the current version that shipped was obviously never used or tested to begin with. It simply does not work.

I understand asking for tests, but it makes little sense for the one line change to turn into what it has been.

**Comment by tpowell-progress** [comment]
@tpowell-progress Something is broken with the verify pipeline... checking in with build team first before merging this in case there's something broken that I'm not seeing.

**Review by jaymzh — APPROVED** [review]
Assuming tests (other than the centos6 buildkite ones) pass.

**Pull request merged**
